packages/expect-utils/src/utils.ts

```diff
@@ -178,7 +192,12 @@ export const iterableEquality = (
   bStack.push(b);
 
   const iterableEqualityWithStack = (a: any, b: any) =>
-    iterableEquality(a, b, [...aStack], [...bStack]);
+    iterableEquality(a, b, [...customTesters], [...aStack], [...bStack]);
+
+  customTesters = [
+    ...customTesters.filter(t => t !== iterableEquality),
```

Comment:
why the check? deserver's a code comment i think

Reply:
is it to prevent infinite recursion in case somebody would manage to add the build-in iterableEquality (somehow imported it directly) into the customTesters array in configuration?

Reply:
Yup - except here all testers are passed in the full array of custom testers, including themselves. So currently it is guaranteed you'll be called with yourself in the custom testers array.

I could also do the filtering in the custom testers loop in `equals` so a tester is never called for itself, but I wasn't sure that was worth since not every tester would need this since not every tester is recursive, but maybe it would make the API easier to work with.

Reply:
Ye, not sure about the additional filtering...